provider/kubernetes/kubernetes.go

```diff
@@ -432,6 +440,55 @@ func loadAuthCredentials(namespace, secretName string, k8sClient Client) ([]stri
 	return creds, nil
 }
 
+func getTLSConfigurations(ingress *v1beta1.Ingress, k8sClient Client) ([]*tls.Configuration, error) {
+	var tlsConfigs []*tls.Configuration
+
+	for _, t := range ingress.Spec.TLS {
+		tlsSecret, exists, err := k8sClient.GetSecret(ingress.Namespace, t.SecretName)
+		if err != nil {
+			return nil, fmt.Errorf("secret %s/%s does not exist: %s", ingress.Namespace, t.SecretName, err)
+		}
+		if !exists {
+			return nil, fmt.Errorf("secret %s/%s does not exist", ingress.Namespace, t.SecretName)
+		}
+
+		tlsCrtData, tlsCrtExists := tlsSecret.Data["tls.crt"]
+		tlsKeyData, tlsKeyExists := tlsSecret.Data["tls.key"]
+		if !tlsCrtExists {
+			return nil, fmt.Errorf("secret %s/%s must have two entries named 'tls.crt' and 'tls.key': missing entry 'tls.crt'", ingress.Namespace, t.SecretName)
+		}
+		if !tlsKeyExists {
+			return nil, fmt.Errorf("secret %s/%s must have two entries named 'tls.crt' and 'tls.key': missing entry 'tls.key'", ingress.Namespace, t.SecretName)
+		}
+
+		var hosts []string
+		if len(t.Hosts) > 0 {
+			hosts = t.Hosts
+		} else {
+			// If no hosts are provided the hosts property should default to the wildcard host. To ensure that the
+			// tls config only matches rules in this ingress, add all the used hosts to simulate a wildcard like
+			// behavior.
+			for _, r := range ingress.Spec.Rules {
+				if r.Host != "" {
+					hosts = append(hosts, r.Host)
+				}
+			}
+		}
+
+		tlsConfig := &tls.Configuration{
+			EntryPoints: hosts,
```

Comment:
Not sure this is correct. The entrypoints are defined in the static config.

Reply:
You are right. I've mixed up traefik's entrypoints with kubernetes ones, which are the domains on which the cert should be used. But as far as I can see there is no way to tell traefik to use a cert for only a set of domains.

Reply:
The easiest way to addres this is probably to check if the provided cert matches the hostnames and isn't a wildcard